### PR TITLE
Split the Wildfy/EAP Maven plugin configuration (from G:A:V to manifest artifact)

### PR DESCRIPTION
**Pull request by fabiobrz**
## Description
This PPR is a follow up to #121 

The required WildFly/EAP 8 Maven plugin configuration needs channel manifest, and this must be reflected by configutration properties.

The PR also refactors the location of such aplication deployments properties so that it's now more straightforward and easy to use.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] Pull Request contains a description of the changes
 - [x] Pull Request does not include fixes for multiple issues/topics
 - [x] Code is self-descriptive and/or documented
 - [x] I have implemented unit tests to cover my changes
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/WildflyDeploymentApplicationConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+ * by the WILDFLY s2i v2 process, i.e. the WildFly Maven plugin, feature packs and BOMs settings
+ */
+public interface WildflyDeploymentApplicationConfiguration {
+
+	String WILDFLY_EE_FEATURE_PACK_LOCATION = "wildfly.ee-feature-pack.location";
+	String WILDFLY_FEATURE_PACK_LOCATION = "wildfly.feature-pack.location";
+	String WILDFLY_CLOUD_FEATURE_PACK_LOCATION = "wildfly.cloud-feature-pack.location";
+	String WILDFLY_EE_CHANNEL_GROUPID = "wildfly.ee-channel.groupId";
+	String WILDFLY_EE_CHANNEL_ARTIFACTID = "wildfly.ee-channel.artifactId";
+	String WILDFLY_EE_CHANNEL_VERSION = "wildfly.ee-channel.version";
```
Comment (tommaso-borgato) [inline deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/WildflyDeploymentApplicationConfiguration.java:33]: perfectly aligned with CI where, on invocation, the `TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST` is splitted:

```shell
# e.g. org.jboss.eap.channels:eap-8.0:1.0.1.GA-redhat-20231212
if [[ "x" != "x$TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST" ]]; then
  arrWfChannelManifestGav=(${TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST//:/ })
  if [[ "x" == "x${arrWfChannelManifestGav[0]}" ]] || [[ "x" == "x${arrWfChannelManifestGav[1]}" ]] || [[ "x" == "x${arrWfChannelManifestGav[2]}" ]]; then
    echo "[ERROR] Invalid value for parameter TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST=$TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST"
    exit 1
  fi
  if [[ ! $ADDITIONAL_MAVEN_ARGS =~ '-Dwildfly.ee-channel.groupId=' ]]; then
    ADDITIONAL_MAVEN_ARGS="$ADDITIONAL_MAVEN_ARGS -Dwildfly.ee-channel.groupId=${arrWfChannelManifestGav[0]}"
  fi
  if [[ ! $ADDITIONAL_MAVEN_ARGS =~ '-Dwildfly.ee-channel.artifactId=' ]]; then
    ADDITIONAL_MAVEN_ARGS="$ADDITIONAL_MAVEN_ARGS -Dwildfly.ee-channel.artifactId=${arrWfChannelManifestGav[1]}"
  fi
  if [[ ! $ADDITIONAL_MAVEN_ARGS =~ '-Dwildfly.ee-channel.version=' ]]; then
    ADDITIONAL_MAVEN_ARGS="$ADDITIONAL_MAVEN_ARGS -Dwildfly.ee-channel.version=${arrWfChannelManifestGav[2]}"
  fi
fi
```

File: deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/WildflyDeploymentApplicationConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		return XTFConfig.get(WILDFLY_MAVEN_PLUGIN_ARTIFACTID);
+	}
+
+	default String wildflyMavenPluginVersion() {
+		return XTFConfig.get(WILDFLY_MAVEN_PLUGIN_VERSION);
+	}
+
+	default String eeFeaturePackLocation() {
+		return XTFConfig.get(WILDFLY_EE_FEATURE_PACK_LOCATION);
+	}
```
Comment (tommaso-borgato) [inline deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/WildflyDeploymentApplicationConfiguration.java:54]: nice idea: IIUC, the former `String eeFeaturePackLocation();` now becomes a default method...

**Review by tommaso-borgato — COMMENTED** [review]
@fabiobrz LGTM, even though parameters were already split, this is a nice refactoring on how to pick them up with defaults methods instead of overriding them each time...

File: deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/WildflyDeploymentApplicationConfiguration.java (hunk 2 above)
Comment (fabiobrz) [reply deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/WildflyDeploymentApplicationConfiguration.java:54]: yup, it is just a configuration interface that behaves in the Intersmash documented way, and we use it easily this way, without the need for overriding that stuff each time.

**Pull request merged**